### PR TITLE
Ensure there are no unused imports with Spotless rather than Checkstyle

### PR DESCRIPTION
**Pull request by basil**
Using [Checkstyle](https://checkstyle.sourceforge.io/) to ensure that there are no unused imports is not ideal because Checkstyle offers no way to automatically fix violations. In contrast, [Spotless](https://github.com/diffplug/spotless) offers both a `check` goal to report violations as well as an `apply` goal to automatically fix all violations, which is far more convenient.

In this PR we disable the [`UnusedImports`](https://checkstyle.sourceforge.io/config_imports.html#UnusedImports) Checkstyle check and replace it with the `removeUnusedImports` Spotless check. The Spotless `check` goal runs in the `verify` phase by default and fails like this if there are any violations:

```
[ERROR] Failed to execute goal com.diffplug.spotless:spotless-maven-plugin:2.10.3:check (default) on project jenkins-test: The following files had format violations:
[ERROR]     src/test/java/jenkins/widgets/HistoryPageFilterCaseSensitiveSearchTest.java
[ERROR]         @@ -3,7 +3,6 @@
[ERROR]          import·java.io.IOException;
[ERROR]          import·java.util.Arrays;
[ERROR]          import·java.util.Collections;
[ERROR]         -import·java.util.List;
[ERROR]          
[ERROR]          import·org.junit.Assert;
[ERROR]          import·org.junit.Rule;
[ERROR] Run 'mvn spotless:apply' to fix these violations.
```

Note that the last line contains a helpful suggestion to run `mvn spotless:apply`, which automatically fixes all violations. The developer can then commit the result and quickly move on.

### CODE REVIEW

**Comment by daniel-beck** [comment]
> Note that the last line contains a helpful suggestion to run `mvn spotless:apply`, which automatically fixes all violations. The developer can then commit the result and quickly move on.

This looks pretty nice 👍

File: pom.xml
Comment (oleg-nenashev) [inline pom.xml:531]: Document in https://github.com/jenkinsci/jenkins/blob/master/CONTRIBUTING.md ? Maybe we need and entry for the quickest-possible build there

**Review by oleg-nenashev — APPROVED** [review]

**Review by res0nance — APPROVED** [review]

File: pom.xml
Comment (basil) [reply pom.xml]: Done

File: CONTRIBUTING.md
Comment (timja) [inline CONTRIBUTING.md:36]: should probably just move to the quick-build profile?
https://github.com/jenkinsci/pom/blob/master/pom.xml#L899

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
